src/url-setters.cpp

```diff
@@ -151,70 +151,12 @@ namespace ada {
     return true;
   }
 
-  bool url::set_hostname(const std::string_view input) {
-    if (has_opaque_path) { return false; }
-
-    std::optional<std::string> previous_host = host;
-
-    std::string_view::iterator input_pointer_end = std::find(input.begin(), input.end(), '#');
-    std::string _host(input.data(), std::distance(input.begin(), input_pointer_end));
-    helpers::remove_ascii_tab_or_newline(_host);
-    std::string_view new_host(_host);
-
-    // If url's scheme is "file", then set state to file host state, instead of host state.
-    if (get_scheme_type() != ada::scheme::type::FILE) {
-      std::string_view host_view(_host.data(), _host.length());
-      bool inside_brackets{false};
-      size_t location = helpers::get_host_delimiter_location(*this, host_view, inside_brackets);
-      std::string_view::iterator pointer = (location != std::string_view::npos) ? new_host.begin() + location : new_host.end();
-
-      // Otherwise, if c is U+003A (:) and insideBrackets is false, then:
-      // Note: we cannot access *pointer safely if (pointer == pointer_end).
-      if ((pointer != new_host.end()) && (*pointer == ':') && !inside_brackets) {
-        // If buffer is the empty string, validation error, return failure.
-        return false;
-      }
-      // If url is special and host_view is the empty string, validation error, return failure.
-      else if (host_view.empty() && is_special()) {
-        return false;
-      }
-      // Otherwise, if state override is given, host_view is the empty string,
-      // and either url includes credentials or url’s port is non-null, return.
-      else if (host_view.empty() && (includes_credentials() || port.has_value())) {
-        return true;
-      }
-
-      // Let host be the result of host parsing host_view with url is not special.
-      if (host_view.empty()) {
-        host = "";
-        return true;
-      }
-
-      bool succeeded = parse_host(host_view);
-      if (!succeeded) { host = previous_host; }
-      return succeeded;
-    }
-
-    size_t location = new_host.find_first_of("/\\?");
-    if (location != std::string_view::npos) { new_host.remove_suffix(new_host.length() - location); }
-
-    if (new_host.empty()) {
-      // Set url’s host to the empty string.
-      host = "";
-    }
-    else {
-      // Let host be the result of host parsing buffer with url is not special.
-      if (!parse_host(new_host)) {
-        host = previous_host;
-        return false;
-      }
+  bool url::set_host(const std::string_view input) {
```

Comment:
This is a breaking change. We should not remove `set_host` or `set_hostname` as a function, but keep the API as it is and just simplify by calling a function similar to what you proposed.

Reply:
I think the git diff is making it a little confusing 😅 set_host and set_hostname are just calling set_host_or_hostname now as a helper, I havent removed the setters

Reply:
Yeah my bad! Git diff is quite confusing. Resolving this.